gobblin-metrics/src/main/avro/MetricReport.avro

```diff
@@ -0,0 +1,37 @@
+// Avro Schema for MetricReport objects used by gobblin-metrics
+// Compile with:
+// java -jar /path/to/avro-tools-1.7.7.jar compile -string schema gobblin-metrics/src/main/avro/MetricReport.avro gobblin-metrics/src/main/gen-java
+{
+  "type": "record",
+  "name": "MetricReport",
+  "namespace": "gobblin.metrics",
+  "fields" : [
+    {
+      "name": "tags",
+      "type": {
+        "type": "map",
+        "values": "string"
+      },
+      "doc": "Tags associated with the metrics."
+    },
+    {
+      "name": "timestamp",
+      "type": "long",
+      "doc": "Tags associated with the metrics."
+    },
+    {
+      "name": "metrics",
+      "type": {
+      "type": "array", "items": {
+        "type": "record",
+        "name": "Metric",
+        "fields": [
+          {"name": "name", "type": "string", "doc": "Metric name."},
+          {"name": "value", "type": "double", "doc": "Metric value."}
```

Comment:
I think it's good to use a `union` type here, which should include all primitive Avro types except for `null`.